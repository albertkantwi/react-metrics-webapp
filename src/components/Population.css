.population {
    display: grid;
    grid-template-columns: 1fr 1px 1fr;
    height: 20rem;
    background-color: #4177c2;
  }
  
  .center_line {
    border-left: 2px solid #1f2937;
  }
  
  .population_img_box {
    width: 100%;
    position: relative;
  }
  
  .population_img_box .world_img {
    width: 100%;
    position: absolute;
    top: 20px;
    right: 5px;
    fill: #52525b;
    margin-top: 2rem;
  }
  
  .population_text {
    display: flex;
    flex-direction: column;
    color: #fff;
    margin: 5.4rem 2rem 0 3.5rem;
  }
  
  .population_text h1 {
    font-size: 3.2rem;
    font-weight: 700;
  }
  
  .population_text p {
    font-size: 1.6rem;
    font-weight: 400;
    line-height: 2.4rem;
  }
  