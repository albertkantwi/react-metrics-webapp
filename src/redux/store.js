import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './country/countrySlice';
import filtersReducer from './searchbox/searchboxSlice';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
    filters: filtersReducer,
  },
});

export default store;
