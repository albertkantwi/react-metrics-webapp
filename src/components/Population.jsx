import React from 'react';
import { useSelector } from 'react-redux';
import './Population.css';

const Population = () => {
  const countries = useSelector((state) => state.country.countriesData);
  const totalPopulation = () => {
    const total = countries.map((country) => country.population).reduce((a, b) => a + b);
    const formatter = total.toLocaleString('en-US');
    return formatter;
  };

  return (
    <div className="population">
      <div className="population_img_box">
        <img className="world_img" src="https://svgsilh.com/svg/306338.svg" alt="World Map" />
      </div>
      <div className="center_line" />
      <div className="population_text">
        <h1>Population</h1>
        <p>
          {countries.length > 0 ? totalPopulation() : 'Loading, please wait!'}
          {' '}
        </p>
      </div>
    </div>
  );
};

export default Population;
