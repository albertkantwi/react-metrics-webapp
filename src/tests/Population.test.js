import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Population from '../components/Population';

test('Check if the Population component is rendered correctly', () => {
  const tree = render(
    <Provider store={store}>
      <BrowserRouter>
        <Population />
      </BrowserRouter>
    </Provider>,
  );

  expect(tree).toMatchSnapshot();
});
