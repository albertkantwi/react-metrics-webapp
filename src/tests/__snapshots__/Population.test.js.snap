// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the Population component is rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="population"
      >
        <div
          class="population_img_box"
        >
          <img
            alt="World Map"
            class="world_img"
            src="https://svgsilh.com/svg/306338.svg"
          />
        </div>
        <div
          class="center_line"
        />
        <div
          class="population_text"
        >
          <h1>
            Population
          </h1>
          <p>
            Loading, please wait!
             
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="population"
    >
      <div
        class="population_img_box"
      >
        <img
          alt="World Map"
          class="world_img"
          src="https://svgsilh.com/svg/306338.svg"
        />
      </div>
      <div
        class="center_line"
      />
      <div
        class="population_text"
      >
        <h1>
          Population
        </h1>
        <p>
          Loading, please wait!
           
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Population Component Check if the Population component is rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="population"
      >
        <div
          class="population_img_box"
        >
          <img
            alt="World Map"
            class="world_img"
            src="https://svgsilh.com/svg/306338.svg"
          />
        </div>
        <div
          class="center_line"
        />
        <div
          class="population_text"
        >
          <h1>
            Population
          </h1>
          <p>
            74,915,796
             
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="population"
    >
      <div
        class="population_img_box"
      >
        <img
          alt="World Map"
          class="world_img"
          src="https://svgsilh.com/svg/306338.svg"
        />
      </div>
      <div
        class="center_line"
      />
      <div
        class="population_text"
      >
        <h1>
          Population
        </h1>
        <p>
          74,915,796
           
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
